pkg/pki/ssh/ssh.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright © 2021 Dan Lorenc <lorenc.d@gmail.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ssh
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"golang.org/x/crypto/ssh"
+)
+
+type Signature struct {
+	signature *ssh.Signature
+	pk        ssh.PublicKey
+}
+
+// NewSignature creates and Validates an ssh signature object
+func NewSignature(r io.Reader) (*Signature, error) {
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+	sig, pk, err := Decode(string(b))
```

Comment:
nit: why do `Armor` and `Decode` return/take string arguments when it appears the first thing you do is convert back to `[]byte`?

Reply:
also, should Decode just return the `Signature` struct as defined on line 28?